src/components/SharedMain/SharedMain.js

```diff
@@ -0,0 +1,39 @@
+import './SharedMain.css';
+import FolderInfo from '../FolderInfo/FolderInfo';
+import SearchBar from '../atoms/SearchBar/SearchBar';
+import CardList from '../CardList/CardList';
+import { getFolderData } from '../../utils/api';
+import { useEffect, useState } from 'react';
+
+export default function SharedMain() {
+  const [folderData, setFolderData] = useState();
+  const [isLoadingSuccess, setIsLoadingSuccess] = useState(false);
+
+  const handleLoadFolderData = async () => {
+    let result;
+    try {
+      result = await getFolderData();
+    } catch (error) {
+      console.log(error);
+    }
+
+    const { folder } = result;
+    setFolderData(folder);
+    setIsLoadingSuccess(true);
+    console.log('폴더 데이터를 가져왔습니다.');
+  };
+
+  useEffect(() => {
+    handleLoadFolderData();
+  }, []);
+
+  return isLoadingSuccess ? (
+    <main>
+      <FolderInfo owner={folderData.owner} folderName={folderData.name} />
+      <article className="test">
+        <SearchBar />
+        <CardList links={folderData.links} />
+      </article>
+    </main>
+  ) : null;
```

Comment:
이런 경우에는 아래처럼 작성해주는 방법도 있을거 같네요~

```suggestion
  if (!isLoadingSuccess) return null

  return  (
   <main>
      <FolderInfo owner={folderData.owner} folderName={folderData.name} />
      <article className="test">
        <SearchBar />
        <CardList links={folderData.links} />
      </article>
    </main>
   )
```

Reply:
이부분도 에러 처리부분이 익숙하지 않아 임시로 작성한 것 같습니다! 좀더 좋은 방법을 고려해보겠습니다!